app/merchant/views.py

```diff
@@ -247,8 +278,47 @@ def view_orders():
     else:
         status_filter = None
 
+    ratings = Ratings.query.filter_by(merchant_id=current_user.id).all()
+    rating_dict = {rating.vendor_id: rating for rating in ratings}
+
     return render_template(
         'merchant/orders.html',
         orders=orders.all(),
-        status_filter=status_filter
+        status_filter=status_filter,
+        ratings=rating_dict
     )
+
+
+@merchant.route('/orders/<int:order_id>', methods=['POST'])
+@login_required
+@merchant_required
+def review_orders(order_id):
+    order = Order.query.get(order_id)
+    star_rating = request.json['rating']
+    comment = request.json['review']
+
+    # TODO: flash on orders page
```

Comment:
"It would be better to remake the vendor function as a form."  - I believe you mean vendor?
So in general you mean that instead of creating the form on the template as it is right now, create a form under merchant to pass to the template and then display the flash on the form?